### PR TITLE
Package bitgenerators.v0.1.0

### PR DESCRIPTION
**Pull request by zoj613**
### `bitgenerators.v0.1.0`
PRNG bitgenerators for OCaml users
A purely functional port of numpy's random bitgenerator interface for OCaml users

---
* Homepage: https://github.com/zoj613/bitgenerators
* Source repo: git+https://github.com/zoj613/bitgenerators.git
* Bug tracker: https://github.com/zoj613/bitgenerators/issues

---
:camel: Pull-request generated by opam-publish v2.3.0

### CODE REVIEW

**Comment by zoj613** [comment]
@mseri How do I resolve the error at https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/d4b0304f0e5315d648198cb00e55a664287beb22/variant/compilers,4.14,bitgenerators.v0.1.0 ? It seems to be caused by CI attempting to build the `bin` directory. Is there a way to skip building it since it's not part of the package release?

**Comment by mseri** [comment]
If you drop `(public_name bench)` and `(public_name crush)` from the [dune file](https://github.com/zoj613/bitgenerators/blob/main/bin/dune), it should not try to compile them unless explicitly requested (e.g. via `dune buil @all` or `dune exec ./bin/crush.exe`)

**Comment by zoj613** [comment]
> If you drop `(public_name bench)` and `(public_name crush)` from the [dune file](https://github.com/zoj613/bitgenerators/blob/main/bin/dune?rgh-link-date=2024-05-01T19%3A05%3A06Z), it should not try to compile them unless explicitly requested (e.g. via `dune buil @all` or `dune exec ./bin/crush.exe`)

Thanks for the response. On my second try to publish I got failures on 32bit Operating systems: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/1c01e9aa882d6ab815d8d31af188d598d457aee1/variant/extras,arm32-ocaml-4.14,bitgenerators.v0.1.0 . Is there a way to skip builds on such machines?

**Comment by zoj613** [comment]
@mseri should I replace `public_name` with `name`? Leaving it out all together returns error: `Error: either the name or the public_name field must be present`

**Comment by mseri** [comment]
Oh yes, indeed

**Comment by zoj613** [comment]
> Oh yes, indeed

Thanks, that worked! How do I add a rule for only building this package on 64bit operating systems? The build fails 32bit platforms.

File: packages/bitgenerators/bitgenerators.v0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/zoj613/bitgenerators.git"
```
Comment (mseri) [inline packages/bitgenerators/bitgenerators.v0.1.0/opam:37]:
```suggestion
dev-repo: "git+https://github.com/zoj613/bitgenerators.git"
conflicts: [ "ocaml-option-bytecode-only" ]
```

Comment (mseri) [reply packages/bitgenerators/bitgenerators.v0.1.0/opam:37]: This should fix the 32 bit issue

Comment (zoj613) [reply packages/bitgenerators/bitgenerators.v0.1.0/opam:37]: Unfortunately, that does not seem to fix the 2 failures on arm32 and x86_32. See here: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/5229ac482af57d7b360a851ffc1048f12c0d7966

**Pull request closed without merging**